pandas/io/sql.py

```diff
@@ -1684,14 +1700,20 @@ def _query_iterator(
         cursor, chunksize, columns, index_col=None, coerce_float=True, parse_dates=None
     ):
         """Return generator through chunked result set"""
+        has_read_data = False
```

Comment:
you could try defining data in an out scope e.g.
```
data: List
```

then you can test that after the break

Reply:
Thank you Jeff.

If I understand what you're proposing correctly (i.e. defining `data` outside the while loop then checking it's status after exiting the loop), then that unfortunately doesn't work. The break clause is triggered when `not data` is True (i.e. `data` is an empty list), so regardless of how many results are returned (and how many times the loop is executed) when you exit the loop `data` will be an empty list. Line 1705 `data = cursor.fetchmany(chunksize)` is always executed at least once, so setting `data` to `None` (or anything else) outside the while loop doesn't achieve anything.